BuildScript/BuildTasks/CreateTestConfigurations.cs

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership.  rubicon licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.  You may obtain a copy of the
+// License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Remotion.BuildScript.BuildTasks
+{
+  public class CreateTestConfigurations : Task
+  {
+    private static readonly MetadataValueCollection s_supportedConfigurationIDs = new MetadataValueCollection (new[] { "debug", "release" });
+
+    [Required]
+    public ITaskItem[] Input { get; set; }
+
+    [Required]
+    public ITaskItem[] SupportedBrowsers { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledBrowsers { get; set; }
+
+    [Required]
+    public ITaskItem[] SupportedDatabaseSystems { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledDatabaseSystems { get; set; }
+
+    [Required]
+    public ITaskItem[] SupportedPlatforms { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledPlatforms { get; set; }
+
+    [Required]
+    public ITaskItem[] SupportedExecutionRuntimes { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledExecutionRuntimes { get; set; }
+
+    [Required]
+    public ITaskItem[] SupportedTargetRuntimes { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledTargetRuntimes { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledConfigurationIDs { get; set; }
+
+    [Output]
+    public ITaskItem[] Output { get; set; }
+
+    public override bool Execute ()
+    {
+      var testConfigurationsFactory = new TestConfigurationsFactory (
```

Comment:
okay, we get them all twice, looks like. introduce variables to remove the duplication. Right now, we actually do have business logic code duplication here due to the specific injection of additional values. Also, there a (perhaps negligible, perhaps not) performance impact due to the dictionary and list building that happens twice now.

Reply:
100% agree. Would you suggest I put the `enabled` ones in variables too? Since there's no duplication with them but we perhaps still want the calls to be symmetric?